### PR TITLE
Set up two more jobs to handle staging and production cases for automerge workflow

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details
This PR does two things:
1. Adds logic for auto merge PRs via the following logic:
- Never automerge for anyone except `OSBotify`
- If the PR target is `master`, only merge if only those specific files are changed, and none are added or deleted
- If the PR target is `staging`, only merge if the PR head branch is `master`
- If the PR target is `production`, only merge if the PR head branch is `staging`

2. Skips all PR checks for auto merge PRs as we merge without waiting for those anyways

### Fixed Issues
Fixes deploys from not running

### Tests
Merge this PR and verify that a staging PR is created and merged automatically

### CODE REVIEW

**Comment by AndrewGable** [comment]
Don't merge, looks like my logic for branches-ignore is wrong.

**Comment by AndrewGable** [comment]
Ok ready for review

File: .github/workflows/preDeploy.yml [hunk 1]
```diff
@@ -38,9 +38,7 @@ jobs:
   skipDeploy:
     runs-on: ubuntu-latest
     needs: chooseDeployActions
-    if: |
-      ${{ needs.chooseDeployActions.outputs.isStagingDeployLocked == 'true' }} &&
-      ${{ needs.chooseDeployActions.outputs.isVersionBumpPR == 'false' }}
+    if: ${{ needs.chooseDeployActions.outputs.isStagingDeployLocked == 'true' }} && ${{ needs.chooseDeployActions.outputs.isVersionBumpPR == 'false' }}
```
Comment (roryabraham) [inline .github/workflows/preDeploy.yml:41]: I think we need this to be like this:

```suggestion
    if: ${{ needs.chooseDeployActions.outputs.isStagingDeployLocked == 'true' && needs.chooseDeployActions.outputs.isVersionBumpPR == 'false' }}
```

File: .github/workflows/preDeploy.yml [hunk 2]
```diff
@@ -140,9 +138,7 @@ jobs:
   updateStaging:
     runs-on: ubuntu-latest
     needs: chooseDeployActions
-    if: |
-      ${{ needs.chooseDeployActions.outputs.isVersionBumpPR == 'true' }} &&
-      ${{ needs.chooseDeployActions.outputs.isStagingDeployLocked == 'false' }}
+    if: ${{ needs.chooseDeployActions.outputs.isVersionBumpPR == 'true' }} && ${{ needs.chooseDeployActions.outputs.isStagingDeployLocked == 'false' }}
```
Comment (roryabraham) [inline .github/workflows/preDeploy.yml:141]:
```suggestion
    if: ${{ needs.chooseDeployActions.outputs.isVersionBumpPR == 'true' && needs.chooseDeployActions.outputs.isStagingDeployLocked == 'false' }}
```

File: .github/workflows/preDeploy.yml [hunk 3]
```diff
@@ -153,6 +149,10 @@ jobs:
       - name: Checkout master branch
         run: git checkout master
 
+      - name: Set Staging Version
+        run: |
```
Comment (roryabraham) [inline .github/workflows/preDeploy.yml:153]: NAB no need for the | here AFAIK

**Review by roryabraham — CHANGES_REQUESTED** [review]
Just a few syntax changes

**Review by roryabraham — APPROVED** [review]
Okay, LGTM

**Comment by botify** [comment]
@roryabraham looks like this was merged without passing tests. Please add a note explaining why this was done or remove the `Emergency` label if this is not an emergency.

**Comment by roryabraham** [comment]
Merging to un-break deploys

**Comment by roryabraham** [comment]
I believe this PR causes duplicate CLA actions to run on pull requests:

image

**Pull request merged**